src/configuration/SSH/OpenSSH/7.3/sshd_config

```diff
@@ -0,0 +1,118 @@
+#ListenAddress 192.168.23.42
+
+# Specifies a file containing a private host key used by SSH.
+# The default is
+## /etc/ssh/ssh_host_dsa_key,
+## /etc/ssh/ssh_host_ecdsa_key,
+## /etc/ssh/ssh_host_ed25519_key and
+## /etc/ssh/ssh_host_rsa_key for protocol version 2.
+HostKey /etc/ssh/ssh_host_ed25519_key
+HostKey /etc/ssh/ssh_host_rsa_key
+
+# Specifies the maximum number of concurrent unauthenticated connections
+# to the SSH daemon.
+# The default is 10:30:100.
+MaxStartups 3
+
+# Specifies the maximum number of authentication attempts permitted per
+# connection.
+# The default is 6.
+MaxAuthTries 1
+
+# Specifies the maximum number of open shell, login or subsystem (e.g. sftp)
+# sessions permitted per network connection.
+# The default is 10.
+MaxSessions 6
+
+#-----------------------------
+# Authentication
+#-----------------------------
+
+# The server disconnects after this time if the user has not
+# successfully logged in.
+# The default is 120 seconds.
+LoginGraceTime 60
+
+# Specifies whether root can log in using ssh(1).
+# The default is "prohibit-password".
+PermitRootLogin no
+
+# Specifies the file that contains the public keys that can be used for
+# user authentication.
+# The default is ".ssh/authorized_keys .ssh/authorized_keys2".
+AuthorizedKeysFile .ssh/authorized_keys
+
+# Specifies whether password authentication is allowed.
+# The default is "yes".
+PasswordAuthentication no
+
+# Specifies whether challenge-response authentication is allowed (e.g.
+# via PAM) or through authentication styles supported in login.conf(5)).
+# The default is "yes".
+ChallengeResponseAuthentication no
+
+# If specified, login is allowed only for user names that match one of
+# the patterns.
+# By default, login is allowed for all users.
+#AllowUsers ed
+
+#-----------------------------
+# Cryptography
+#-----------------------------
+
+# Specifies the available KEX (Key Exchange) algorithms.
+# The default is:
+## curve25519-sha256@libssh.org,
+## ecdh-sha2-nistp256,
+## ecdh-sha2-nistp384,
+## ecdh-sha2-nistp521,
+## diffie-hellman-group-exchange-sha256,
+## diffie-hellman-group14-sha1.
+KexAlgorithms curve25519-sha256@libssh.org,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256
+
+# Specifies the ciphers allowed.
+# The default is:
+## chacha20-poly1305@openssh.com,
+## aes128-ctr,
+## aes192-ctr,
+## aes256-ctr,
+## aes128-gcm@openssh.com,
+## aes256-gcm@openssh.com.
+Ciphers chacha20-poly1305@openssh.com,aes256-ctr,aes256-gcm@openssh.com
+
+# Specifies the available MAC (message authentication code) algorithms.
+# The default is:
+## umac-64-etm@openssh.com,
+## umac-128-etm@openssh.com,
+## hmac-sha2-256-etm@openssh.com,
+## hmac-sha2-512-etm@openssh.com,
+## hmac-sha1-etm@openssh.com,
+## umac-64@openssh.com,
+## umac-128@openssh.com,
+## hmac-sha2-256,
+## hmac-sha2-512,
+## hmac-sha1.
+MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512,hmac-sha2-256
```

Comment:
Sorry, but no. Just because an algorithm has a bigger number at the end doesn't mean it's more secure. You should really spend time on investigating the different choices you make before coming to a decision about security critical algorithm settings.

* encrypt-then-mac vs. mac-and-encrypt (OpenSSH default):
In reality we only want encrypt-then-mac (etm). Some cryptographic attacks work because encrypt-then-mac wasn't used. Here's a good overview on the subject: http://cseweb.ucsd.edu/~mihir/papers/oem.pdf

* UMAC:
https://www.openssh.com/txt/draft-miller-secsh-umac-01.txt
```
 This memo describes the use of the UMAC Message Authentication Code
   [RFC4418] in the SSH transport protocol.  UMAC offers improved
   performance over the current HMAC-based MACs supported by SSH.
   Furthermore, UMAC represents a different cryptographic approach to
   message authentication to that of HMAC, and its use in SSH would
   provide a diversity that may be of benefit if HMAC or one of its
   underlying hash algorithms is found to be vulnerable to a new attack.

   UMAC offers a choice of four authentication tag sizes: 32, 64, 96 and
   128 bits.  This allows users to trade between fast, compact tags and
   longer tags that are slower to generate but offer better security.
   These four tag sizes are represented in the new MAC methods
   introduced in this memo: "umac-32", "umac-64", "umac-96" and "umac-
   128".
```
UMAC is described in detail in [RFC4418](https://tools.ietf.org/html/rfc4418). The underlying block cipher used today is AES. [Section 4](https://tools.ietf.org/html/rfc4418#section-4) of RFC4418:
```
4.  UMAC Tag Generation

   Tag generation for UMAC proceeds by using UHASH (defined in the next
   section) to hash the message, applying the PDF to the nonce, and
   computing the xor of the resulting strings.  The length of the pad
   and hash can be either 4, 8, 12, or 16 bytes.
```

In contrast to a HMAC of any kind (e.g. SHA2), the number at the end (with UMAC) refers to the tag-size not the key-size used as input to the MAC. For SHA2 that'll be 256 or 512 bits. Sometimes truncation is used. On the security HMAC's and their implementation see also [RFC2104](https://tools.ietf.org/html/rfc2104).